broker/provision_endpoint.go

```diff
@@ -49,15 +50,40 @@ func (bkr *Broker) provision(instanceID structs.ClusterID, details brokerapi.Pro
 	// Continue processing in background
 	// TODO: if error, store it into etcd; and last_operation_endpoint should look for errors first
 	go func() {
-		err := bkr.scheduler.RunCluster(clusterModel, features)
-		if err != nil {
+		if err := bkr.scheduler.RunCluster(clusterModel, features); err != nil {
 			logger.Error("run-cluster", err)
 			return
 		}
 
-		err = bkr.router.AssignPortToCluster(clusterModel.InstanceID(), port)
-		if err != nil {
+		if err := bkr.router.AssignPortToCluster(instanceID, port); err != nil {
 			logger.Error("assign-port", err)
+			return
+		}
+
+		// If broker has credentials for a Cloud Foundry,
+		// attempt to look up service instance to get its user-provided name.
+		// This can then be used in future to undo/recreate-from-backup when user
+		// only knows the name they provided; and not the internal service instance ID.
+		// If operation fails, that's temporarily unfortunate but might be due to credentials
+		// not yet having SpaceDeveloper role for the Space being used.
+		if bkr.cf != nil && bkr.callbacks.Configured() {
+			serviceInstanceName, err := bkr.cf.LookupServiceName(instanceID)
+			if err != nil {
+				logger.Error("lookup-service-name.error", err,
+					lager.Data{"action-required": "Fix issue and run errand/script to update clusterdata backups to include service names"})
+			}
+			if serviceInstanceName == "" {
+				logger.Info("lookup-service-name.not-found")
+			} else {
```

Comment:
The recreation data is created originally earlier in the provision method in case of any subsequent failures. This is a 2nd save/update. That ok?